src/EventStore.Core/ProcessTracker.cs

```diff
@@ -0,0 +1,221 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Reflection;
+using System.Threading;
+using EventStore.Core.Services.Monitoring.Stats;
+using EventStore.Core.Services.Monitoring.Utils;
+using EventStore.Core.Time;
+using EventStore.Native.Monitoring;
+using HostStat;
+using Serilog;
+using Exception = System.Exception;
+
+namespace EventStore.Core;
+
+public class ProcessTracker {
+	private static readonly ILogger _log =
+		Serilog.Log.ForContext(Serilog.Core.Constants.SourceContextPropertyName, "PROCESS-TRACKER-LOGGER");
+
+	private readonly HostStat.HostStat _stats;
+	private readonly PerfCounterHelper _perfCounter;
+	private readonly Func<int, long> _gcCollectSize;
+	private readonly Func<int> _exceptionCount;
+	private readonly Func<int> _timeInGc;
+	private readonly Func<int> _procCpuUsage;
+	private readonly Func<Process> _currentProc;
+	private readonly Func<LoadAverages> _loadAverages;
+	private readonly Func<EsDriveInfo?> _esDriveInfo;
+	private readonly Func<DiskIo?> _diskIo;
+
+	public ProcessTracker(IClock clock, string dbPath) {
```

Comment:
what do you think about splitting this up into sys/proc/disk

(with disk used/total bytes in sys i think)

Reply:
Yeah, it will also enables finer tuning on what we want to  collect metrics from.